compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -629,15 +629,19 @@ impl InitMask {
         Size::from_bytes(block * InitMask::BLOCK_SIZE + bit)
     }
 
-    pub fn new(size: Size, state: bool) -> Self {
+    pub fn new_init(size: Size) -> Self {
         let mut m = InitMask { blocks: vec![], len: Size::ZERO };
-        m.grow(size, state);
+        m.grow(size, true);
         m
     }
 
+    pub fn new_uninit(size: Size) -> Self {
+        InitMask { blocks: vec![], len: size }
+    }
+
     pub fn set_range(&mut self, start: Size, end: Size, new_state: bool) {
         let len = self.len;
-        if end > len {
+        if end > len && new_state {
```

Comment:
I am a bit surprised that we allow OOB indices at all here?

Reply:
Also, I think set_range_inbounds is called in situations where the index is in-bounds of the logical size of the InitMap, but OOB of its actual size due to leaving off the tail. So I think the growing logic needs to move to `set_range_inbounds`.

Reply:
Ah, you have `ensure_blocks` for that. But then we have two growing logics? That seems odd.